lib/AST/Decl.cpp

```diff
@@ -1937,15 +1937,15 @@ ValueDecl::getFormalAccessScope(const DeclContext *useDC) const {
   case Accessibility::Private:
   case Accessibility::FilePrivate:
     assert(result->isModuleScopeContext());
-    return result;
+    return AccessScope(result, access == Accessibility::Private);
   case Accessibility::Internal:
-    return result->getParentModule();
+    return AccessScope(result->getParentModule());
   case Accessibility::Public:
   case Accessibility::Open:
-    return nullptr;
+    return AccessScope();
   }
 
-  return result;
+  return AccessScope(result);
```

Comment:
Oh, the above is a fully-covered switch, right? Let's use `llvm_unreachable` here instead. (I'm not sure why it wasn't doing that already.)